lib/screens/second_screen.dart

```diff
@@ -0,0 +1,90 @@
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:dio/dio.dart';
+import 'package:networking/data/album.dart';
+
+class SecondScreen extends StatelessWidget {
+  const SecondScreen({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("SecondScreen"),
+      ),
+      body: FutureBuilder<List<dynamic>?>(
+          future: getHttp(),
+          builder: (BuildContext context, AsyncSnapshot<List?> list) {
+            Widget child;
+            if (list.hasData) {
+              child = ListAlbums(list: list);
+            } else if (list.hasError) {
+              child = Center(
+                child: Text("${list.error}"),
+              );
+            } else {
+              child = const Center(
+                child: CircularProgressIndicator(),
+              );
+            }
+            return child;
+          }),
+    );
+  }
+
+  Future<List<Album>?> getHttp() async {
+    try {
+      Response response =
+          await Dio().get('https://jsonplaceholder.typicode.com/posts');
+      if (response.statusCode == 200) {
+        var albumData = response.data as List;
+        var listAlbums =
+            albumData.map((element) => Album.fromJson(element)).toList();
+        return listAlbums;
+      } else {
+        throw Exception("Failed to load albums"); //в реальных приложениях выкидывают ексепшены?
```

Comment:
Please see my comment [here](https://github.com/AlexMeshcharakou/flutter_network/pull/1#discussion_r831734660).

In the real app, you need some error message or errors screen, often it contains the retry button to reload the data. In our case to simplicity, you can add just a simple screen or popup or snake bar with the text "Ooops, we cannot get this data right now"

Reply:
change flutter channel to stable. Add state processing in FutureBuilder.